### PR TITLE
update sanity check in OpenMPI easyblock to support OpenMPI v5.0.0

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyblocks/o/openmpi.py [hunk 1]
```diff
@@ -157,14 +157,24 @@ def load_module(self, *args, **kwargs):
     def sanity_check_step(self):
         """Custom sanity check for OpenMPI."""
 
-        bin_names = ['mpicc', 'mpicxx', 'mpif90', 'mpifort', 'mpirun', 'ompi_info', 'opal_wrapper', 'orterun']
+        bin_names = ['mpicc', 'mpicxx', 'mpif90', 'mpifort', 'ompi_info', 'opal_wrapper']
+        if LooseVersion(self.version) >= LooseVersion('5.0.0'):
+            bin_names.append('prterun')
+        else:
+            bin_names.extend(['mpirun', 'orterun'])
```
Comment (bartoldeman) [inline easybuild/easyblocks/o/openmpi.py:164]: just curious: what happened to `mpirun` in 5.0.0?

**Review by bartoldeman — APPROVED** [review]

**Review by bartoldeman — DISMISSED** [review]
sorry I should request changes.

We can't reasonably expect people to run prterun instead of mpirun, so mpirun must be somewhere, and there are are some issues on Open MPI's repo about mpirun so it hasn't disappeared.

File: easybuild/easyblocks/o/openmpi.py (hunk 1 above)
Comment (robert-mijakovic) [reply easybuild/easyblocks/o/openmpi.py:164]: Sorry, I made a mistake.
In 4.x, if you would use PRRTE, mpirun would not be available.
However, it reappeared in 5.0.x.

**Comment by boegel** [comment]
> sorry I should request changes.
> 
> We can't reasonably expect people to run prterun instead of mpirun, so mpirun must be somewhere, and there are are some issues on Open MPI's repo about mpirun so it hasn't disappeared.

I remember soemthing about `mpirun` going away and being replaced by `prterun`, so maybe they did indeed remove it in the end?
Surely this is mentioned in the (draft) OpenMPI v5.0.0 release notes if that's indeed the case...

**Comment by robert-mijakovic** [comment]
@boegel @bartoldeman The issue is addressed.

File: easybuild/easyblocks/o/openmpi.py [hunk 2]
```diff
@@ -124,11 +124,11 @@ def config_opt_used(key, enable_opt=False):
                 # auto-detect based on available OS packages
                 os_packages = EASYCONFIG_CONSTANTS['OS_PKG_IBVERBS_DEV'][0]
                 verbs = any(check_os_dependency(osdep) for osdep in os_packages)
-
-            if verbs:
-                self.cfg.update('configopts', '--with-verbs')
-            else:
-                self.cfg.update('configopts', '--without-verbs')
+            if LooseVersion(self.version) <= LooseVersion('5.0.0'):
```
Comment (boegel) [inline easybuild/easyblocks/o/openmpi.py:127]: @robert-mijakovic Can you clarify this? If `--with-verbs` no longer valid in OpenMPI 5.0.0?

How should we explicitely disable/enable IB support then for OpenMPI >= 5.0.0?

Comment (akesandgren) [reply easybuild/easyblocks/o/openmpi.py:127]: IB support in 5 comes only from UCX if I remember correctly??

Comment (robert-mijakovic) [reply easybuild/easyblocks/o/openmpi.py:127]: @akesandgren Yes, that is my impression too.

Comment (boegel) [reply easybuild/easyblocks/o/openmpi.py:127]: OK, let's add a comment above this `if` with a reference to https://github.com/open-mpi/ompi/pull/6270, just so we're not surprised by this going forward.

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _OpenMPI-3.1.4-GCC-8.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.0.3-GCC-9.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.0.5-GCC-10.2.0.eb_ 
 * **SUCCESS** _OpenMPI-4.1.1-GCC-10.3.0.eb_ 
 * **SUCCESS** _OpenMPI-4.1.1-GCC-11.2.0.eb_ 

Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3106.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/0e8b3b88ced44ebca8fa4329e515c795 for a full test report.

**Review by boegel — APPROVED** [review]

**Pull request merged**